shopelectro/views/catalog.py

```diff
@@ -25,6 +26,62 @@ def get_products_count(request):
     return PRODUCTS_ON_PAGE_MOB if mobile_view else PRODUCTS_ON_PAGE_PC
 
 
+def get_view_type(request):
+    view_type = request.session.get('view_type', 'tile')
+    assert view_type in ['list', 'tile']
+    return view_type
+
+
+def get_catalog_context(request, category, raw_tags, page_number, per_page, sorting_index):
+    all_tags = newcontext.Tags(models.Tag.objects.all())
+
+    if raw_tags:
+        selected_tags = newcontext.tags.Checked404Tags(
```

Comment:
`Checked404Tags` looks confusing. Maybe `ExistingTags` or smth?

Reply:
Is it look ok If we will rename it to `ExistOr404`?

Reply:
@artemiy312 , then `ExistingOr404`. It'll be good to have nouns in class names.
But decide it